BugSnag ↦ Sentry

rye opened the pull request:
This PR deletes the integrations with BugSnag and adds the barebones integrations with Sentry. We now no longer have a `notify` function, as I anticipated using Sentry's provided methods over that.  We can always write some helpers if we want.

Still yet to figure out:
- [x] Source map + dSYM uploading (from CI)
  - [x] Need to include `auth-token` in `sentry.properties`
- [ ] Git integration

rye committed:
Remove BugSnag from our project

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Add react-native-sentry and @sentry/cli as dependencies

All in all, not that big of a hit size-wise.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Link react-native-sentry

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Set up Sentry client

We can interact with it by talking to the global `Sentry` constant,
which is nice.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives commented on android/sentry.properties:
```diff
@@ -0,0 +1,3 @@
+defaults.url=https://sentry.frogpond.tech/
+defaults.org=frog-pond-labs
+defaults.project=all-about-olaf
```
One thought - could we include a commented out version of the API key line in both of these files?

rye replied:
Yeah, we could.

rye committed:
Resolve a lint error

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -5,7 +5,6 @@
 	};
 	objectVersion = 46;
 	objects = {
-
```
Keep this line [xcode]

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -1100,7 +1094,7 @@
 			isa = PBXProject;
 			attributes = {
-				LastSwiftUpdateCheck = 0810;
+				LastSwiftUpdateCheck = 810;
```
Keep the 0 [xcode]

rye committed:
CHANGELOG

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -1111,7 +1105,7 @@
 					13B07F861A680F5B00A75B9A = {
 						DevelopmentTeam = TMK6S7TPX2;
-						LastSwiftMigration = 0810;
+						LastSwiftMigration = 810;
```
Keep the 0 [xcode]

hawkrives commented on ios/AllAboutOlaf/AppDelegate.m:
```diff
@@ -14,6 +14,11 @@
+#import <React/RNSentry.h> // This is used for versions of react >= 0.40
+#else
+#import "RNSentry.h" // This is used for versions of react < 0.40
+#endif
```
Remove the else branch

hawkrives replied:
And by proxy, leave only the content of the if - there's no need for the conditional here at all, since we're running 0.59

rye replied:
Okay, yeah.

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -1730,7 +1713,7 @@
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "export NODE_BINARY=node\n../node_modules/react-native/scripts/react-native-xcode.sh";
+			shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport NODE_BINARY=node\n../node_modules/@sentry/cli/bin/sentry-cli react-native xcode ../node_modules/react-native/scripts/react-native-xcode.sh";
```
Something tells me that we need another \n in here, or we'll wind up running "sentry-cli React-native" which I don't think is a command

rye replied:
This was autogenerated, and I don't think it should be a problem.

hawkrives replied:
I'm game - if CI builds, it's not a problem here

hawkrives commented on source/init/sentry.js:
```diff
@@ -0,0 +1,14 @@
+		handlePromiseRejection: true,
+	})
+
+	if (!IS_PRODUCTION) config.install()
```
Make this if multi-line please

rye replied:
Prettier did this.

hawkrives replied:
Put braces around it and prettier will leave it multi-line

hawkrives commented on modules/add-to-device-calendar/lib.js:
```diff
@@ -1,7 +1,6 @@
 // @flow
 
 import type {EventType} from '@frogpond/event-type'
-import {notify} from '@frogpond/analytics'
```
Thinking into the app logic… I see a lot of notify/breadcrumb removals. Do we want to keep those? Does sentry automatically do them?

rye replied:
Sentry "should" handle crashes and the like, I believe. For sure, it includes the necessary beef to handle unhandled promise rejections, but beyond that I am not sure if it will e.g. catch uncaught errors.

rye replied:
Addendum: `notify` was mostly used to explicitly send out an event or error message, and we will need to manually use the Sentry equivalent of that.

hawkrives reviewed (commented):
Some thoughts

rye committed:
ios: Remove else branch for old versions of RCT

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ios: Keep the 0

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ios: Keep the newline

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ios: Remove last references to BugSnag

Apparently `git grep` is case sensitive.  Who knew!

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ios: Remove unnecessary if-guard

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
init/sentry: Make if multi-line

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
android: Don't apply rules from Sentry

We will manually upload sourcemaps because we're cool and we know what
we're doing.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ios: Don't automatically upload Sentry things

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Add some Sentry logging back

Woop woop!

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ios: Remove dangling reference to debug symbol upload

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ios: Remove undefined reference to BSRN

Just one line was hidden deep in the logs, and it took me this long to
to find it.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
fastlane: Add upload_sourcemap_to_sentry

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Add sentry auth token (commented out) to sentry.properties

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
circle: Pass SENTRY_PROPERTIES to the ios and android steps

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives commented on fastlane/lib/sourcemaps.rb:
```diff
@@ -44,18 +44,21 @@ def generate_sourcemap
 
-# Upload source map to Bugsnag
+# Upload sourcemap to bugsnag
 def upload_sourcemap_to_bugsnag
```
Rename this method?

hawkrives commented on source/views/help/wifi.js:
```diff
@@ -50,8 +50,7 @@ export class ToolView extends React.Component<Props, State> {
 		const [position, device] = await Promise.all([
-			getPosition().catch(error => {
-				notify(error)
+			getPosition().catch(() => {
```
Put the notify call back in?

hawkrives reviewed (dismissed):
A few tweaks have been requested by me

rye committed:
fastlane: Rename sourcemap upload (typo)

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
wifi: Add Sentry.captureException call

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

drewvolz reviewed (approved):
👏 good job, @rye! :shipit:
